### PR TITLE
Admin filtering

### PR DESCRIPTION
**Pull request by ashryanbeats**

### CODE REVIEW

File: tests/server/routes/user.spec.js [hunk 1]
```diff
@@ -169,14 +217,24 @@ var cookie=[];
...
+        .end(function(err, res) {
+          if (err) return done(err);
+          done();
+        });
+    });    
+
+    it('`/:id` Can delete self from the db', function(done) {
+      request
+        .delete('/api/user/' + selfId)
         .expect(201)
```
Comment (colin92) [inline tests/server/routes/user.spec.js]: this should be 401 no? 201 is the code for created.

File: tests/server/routes/user.spec.js [hunk 2]
```diff
@@ -134,22 +171,33 @@ var cookie=[];
...
     });
 
-    it('`/:id` Gets a 201 response and updates to the db', function(done) {
+    it('`/:id` Gets a 201 response and updates to the db for self', function(done) {
       request
-        .put('/api/user/' + id)
-        .send(user)
+        .put('/api/user/' + selfId)
+        .send({firstName: "Moe"})
         .expect(201)
```
Comment (colin92) [inline tests/server/routes/user.spec.js]: successful put should be 200 or 204

File: tests/server/routes/user.admin.spec.js [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+          id = savedUser._id;
+          done();
+        })
+        .then(null, done);
+    });
+
+    it('`/:id` Gets a 201 response and deletes from the db', function(done) {
+      request
+        .delete('/api/user/' + id)
+        .expect(201)
```
Comment (colin92) [inline tests/server/routes/user.admin.spec.js]: change 201 to 200 or 204.

File: tests/server/routes/user.admin.spec.js [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+          done();
+        })
+        .then(null, done);
+    });
+
+    it('`/:id` Gets a 201 response and updates to the db', function(done) {
+      request
+        .put('/api/user/' + id)
+        .send(user)
+        .expect(201)
```
Comment (colin92) [inline tests/server/routes/user.admin.spec.js]: change 201 to 200 or 204

File: tests/server/routes/user.spec.js (hunk 1 above)
Comment (ashryanbeats) [inline tests/server/routes/user.spec.js]: Done.

File: tests/server/routes/user.spec.js (hunk 2 above)
Comment (ashryanbeats) [inline tests/server/routes/user.spec.js]: Done.

File: tests/server/routes/user.admin.spec.js (hunk 3 above)
Comment (ashryanbeats) [inline tests/server/routes/user.admin.spec.js]: Done.

File: tests/server/routes/user.admin.spec.js (hunk 4 above)
Comment (ashryanbeats) [inline tests/server/routes/user.admin.spec.js]: Done.

**Pull request merged**